File: api/src/interface/controllers/UserController.ts
import { GetUsers } from "../../application/usecases/GetUsers";
import { User } from "../../domain/entities/User";
import {
  ControllerRequest,
  ControllerResponse,
} from "../../infrastructure/ExpressInterfaceAdapter";

export class UserController {
  public async get(request: ControllerRequest): Promise<ControllerResponse> {
    console.log(request);
    const getUserUsecase = new GetUsers();
    const users: User[] = await getUserUsecase.getUsers();
    const usersResponse: UserResponse[] = [
      this.toUserGetResponse(users[0]),
      this.toUserGetResponse(users[1]),
      this.toUserGetResponse(users[2]),
    ];

    const response: ControllerResponse = {
      status: 200,
      body: usersResponse,
    };
    return response;
  }
  private toUserGetResponse(user: User): UserResponse {
    return {
      id: user.id,
      name: user.name,
      email: user.email,
    };
  }
}

export type UserResponse = {
  id: number;
  name: string;
  email: string;
};

File: flask-api/src/interface/controllers/UserController.py
from src.infrastructure.FlakeInterfaceAdapter import ControllerRequest


class UserResponse:
  def __init__(self, id: int, name: str, email: str) -> None:
    self.id = id
    self.name = name
    self.email = email

class UserController:
  def get(self, request: ControllerRequest) -> list[UserResponse]:
    print("request.__dict__")
    print(request.__dict__)
    users_response = [
      UserResponse(1, "TestUser1", "testuser1@test.com"),
      UserResponse(2, "TestUser2", "testuser2@test.com"),
      UserResponse(3, "TestUser3", "testuser3@test.com")
    ]
    return users_response

